nifi-nar-bundles/nifi-standard-services/nifi-ssl-context-bundle/nifi-ssl-context-service/src/main/java/org/apache/nifi/ssl/StandardSSLContextService.java

```diff
@@ -185,14 +191,24 @@ public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, S
             }
         }
 
-        results.addAll(validateStore(validationContext.getProperties(), KeystoreValidationGroup.KEYSTORE));
-        results.addAll(validateStore(validationContext.getProperties(), KeystoreValidationGroup.TRUSTSTORE));
+        final Map<PropertyDescriptor, String> properties = evaluateProperties(validationContext.getProperties(), validationContext);
+        results.addAll(validateStore(properties, KeystoreValidationGroup.KEYSTORE));
+        results.addAll(validateStore(properties, KeystoreValidationGroup.TRUSTSTORE));
 
         isValidated = results.isEmpty();
 
         return results;
     }
 
+    private Map<PropertyDescriptor, String> evaluateProperties(final Map<PropertyDescriptor, String> properties, final PropertyContext context) {
+        for (final PropertyDescriptor pd : properties.keySet()) {
+            if (pd.isExpressionLanguageSupported() && context.getProperty(pd).isSet()) {
+                properties.put(pd, context.getProperty(pd).evaluateAttributeExpressions().getValue());
```

Comment:
Chris,
I looked into this a little more, and discovered the error is occurring due to an attempt to modify an unmodifiable collection (map) here.
I was able to avoid the UnsupportedOperationException by creating a new HashMap for the evaluated properties, and return it from this method. I haven't fully thought through if this is a concern with effectively side-stepping the "unmodifiable" intent of the map.

    private Map<PropertyDescriptor, String> evaluateProperties(final Map<PropertyDescriptor, String> properties, final PropertyContext context) {
        Map<PropertyDescriptor, String> evaluatedProperties = new HashMap<>();
        for (final PropertyDescriptor pd : properties.keySet()) {
            if (pd.isExpressionLanguageSupported() && context.getProperty(pd).isSet()) {
                evaluatedProperties.put(pd, context.getProperty(pd).evaluateAttributeExpressions().getValue());
            } else {
                evaluatedProperties.put(pd, context.getProperty(pd).getValue());
            }
        }
        return evaluatedProperties;
    }

-Mark

Reply:
@markobean I figured it would be something like that. Have been on holiday this past week, will hopefully get to look at this sometime soon once back

Reply:
Fixed with code similar to what you suggested (I tried using Java Streams initially but the `Collectors.toMap` doesn't like `null` values, so instead I went with a straight up "old school" for` loop)

Reply:
As to the modifiability of the Map - these properties are only used internally within the controller service and (currently) are treated as read-only by all the methods that use them (so far as I can see)

So we could (maybe should) make them unmodifiable after EL has been evaluated (way to do although I forgot to include in my latest commit) - can do this if you think best

The underlying properties (descriptors & values) should not be affected by the nature of the Map used herein